src/Symfony/Bridge/Doctrine/Validator/Constraints/UniqueEntityValidator.php

```diff
@@ -227,4 +269,44 @@ private function formatWithIdentifiers(ObjectManager $em, ClassMetadata $class,
 
         return sprintf('object("%s") identified by (%s)', $idClass, implode(', ', $identifiers));
     }
+
+    public function getFieldValues($object, ClassMetadata $class, array $fields, bool $isEntity = false): array
+    {
+        if (!$isEntity) {
+            $reflectionObject = new \ReflectionObject($object);
+        }
+
+        $fieldValues = [];
+        $objectClass = \get_class($object);
+
+        foreach ($fields as $objectFieldName => $entityFieldName) {
+            if (!$class->hasField($entityFieldName) && !$class->hasAssociation($entityFieldName)) {
+                throw new ConstraintDefinitionException(sprintf('The field "%s" is not mapped by Doctrine, so it cannot be validated for uniqueness.', $entityFieldName));
+            }
+
+            $fieldName = \is_int($objectFieldName) ? $entityFieldName : $objectFieldName;
+            if (!$isEntity) {
+                if (!$reflectionObject->hasProperty($fieldName)) {
+                    throw new ConstraintDefinitionException(sprintf('The field "%s" is not a property of class "%s".', $fieldName, $objectClass));
+                }
+            }
+
+            $fieldValues[$entityFieldName] = $this->getPropertyValue($objectClass, $fieldName, $object);
+        }
+
+        return $fieldValues;
+    }
+
+    /**
+     * @return mixed
+     */
+    public function getPropertyValue(string $class, string $name, $object)
+    {
+        $property = new \ReflectionProperty($class, $name);
+        if (!$property->isPublic()) {
+            $property->setAccessible(true);
+        }
+
```

Comment:
If this property doesn't exist in DTO Reflection will throw an error in getting value from object  `$property->getValue($object);`.
Because property isn't initialized at this moment.
```suggestion

        if (!$property->isInitialized($object)) {
            return null;
        }
        
```

Reply:
Good catch. Added test and fixed.

Reply:
In my opinion, such a situation should not occur and it is an implementation error.
Maybe throwing the `UninitializedPropertyException` is a better option?
What do other reviewers think?